File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/objects/SLObjectAvatarInfo.java
package com.lumiyaviewer.lumiya.slproto.objects;

import com.lumiyaviewer.lumiya.render.avatar.AvatarVisualState;
import com.lumiyaviewer.lumiya.render.spatial.DrawListAvatarEntry;
import com.lumiyaviewer.lumiya.render.spatial.DrawListObjectEntry;
import com.lumiyaviewer.lumiya.slproto.messages.AvatarAnimation;
import com.lumiyaviewer.lumiya.slproto.messages.AvatarAppearance;
import com.lumiyaviewer.lumiya.slproto.textures.SLTextureEntry;
import java.util.UUID;
import javax.annotation.Nonnull;

public class SLObjectAvatarInfo extends SLObjectInfo {
    @Nonnull
    private final AvatarVisualState avatarVisualState;
    private final boolean isMyAvatar;

    public SLObjectAvatarInfo(UUID uuid, UUID uuid2, boolean z) {
        this.isMyAvatar = z;
        this.avatarVisualState = new AvatarVisualState(uuid, this, uuid2);
    }

    public void ApplyAvatarAnimation(AvatarAnimation avatarAnimation) {
        this.avatarVisualState.ApplyAvatarAnimation(avatarAnimation);
    }

    public void ApplyAvatarAppearance(AvatarAppearance avatarAppearance) {
        this.avatarVisualState.ApplyAvatarAppearance(avatarAppearance);
    }

    public void ApplyAvatarTextures(SLTextureEntry sLTextureEntry, boolean z) {
        this.avatarVisualState.ApplyTextures(sLTextureEntry, z);
    }

    public void ApplyAvatarVisualParams(int[] iArr) {
        this.avatarVisualState.ApplyVisualParams(iArr);
    }

    /* access modifiers changed from: protected */
    @Nonnull
    public DrawListObjectEntry createDrawListEntry() {
        return new DrawListAvatarEntry(this);
    }

    @Nonnull
    public AvatarVisualState getAvatarVisualState() {
        return this.avatarVisualState;
    }

    public String getName() {
        return this.isMyAvatar ? "(my avatar)" : "(avatar)";
    }

    public boolean isAvatar() {
        return true;
    }

    public boolean isMyAvatar() {
        return this.isMyAvatar;
    }

    public void onTexturesUpdate(SLTextureEntry sLTextureEntry) {
        this.avatarVisualState.ApplyTextures(sLTextureEntry, false);
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/objects/SLObjectAvatarInfo.kt
package com.lumiyaviewer.lumiya.slproto.objects

import com.lumiyaviewer.lumiya.render.avatar.AvatarVisualState
import com.lumiyaviewer.lumiya.render.spatial.DrawListAvatarEntry
import com.lumiyaviewer.lumiya.render.spatial.DrawListObjectEntry
import com.lumiyaviewer.lumiya.slproto.messages.AvatarAnimation
import com.lumiyaviewer.lumiya.slproto.messages.AvatarAppearance
import com.lumiyaviewer.lumiya.slproto.textures.SLTextureEntry
import java.util.UUID
import javax.annotation.Nonnull

class SLObjectAvatarInfo : SLObjectInfo {
    @Nonnull
    private AvatarVisualState avatarVisualState
    private Boolean isMyAvatar

    SLObjectAvatarInfo(UUID uuid, UUID uuid2, Boolean z) {
        this.isMyAvatar = z
        this.avatarVisualState = AvatarVisualState(uuid, this, uuid2)
    }

    Unit ApplyAvatarAnimation(AvatarAnimation avatarAnimation) {
        this.avatarVisualState.ApplyAvatarAnimation(avatarAnimation)
    }

    Unit ApplyAvatarAppearance(AvatarAppearance avatarAppearance) {
        this.avatarVisualState.ApplyAvatarAppearance(avatarAppearance)
    }

    Unit ApplyAvatarTextures(SLTextureEntry sLTextureEntry, Boolean z) {
        this.avatarVisualState.ApplyTextures(sLTextureEntry, z)
    }

    Unit ApplyAvatarVisualParams(IntArray iArr) {
        this.avatarVisualState.ApplyVisualParams(iArr)
    }

    /* access modifiers changed from: protected */
    @Nonnull
    DrawListObjectEntry createDrawListEntry() {
        return DrawListAvatarEntry(this)
    }

    @Nonnull
    AvatarVisualState getAvatarVisualState() {
        return this.avatarVisualState
    }

    String getName() {
        return this.isMyAvatar ? "(my avatar)" : "(avatar)"
    }

    Boolean isAvatar() {
        return true
    }

    Boolean isMyAvatar() {
        return this.isMyAvatar
    }

    Unit onTexturesUpdate(SLTextureEntry sLTextureEntry) {
        this.avatarVisualState.ApplyTextures(sLTextureEntry, false)
    }
}
